components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1181,12 +1183,690 @@ private class MonitoredItemWrapper {
                 return fieldName;
             }
 
+            internal EventFilter BuildBasicEventFilter(Session session) {
+                var simpleEventTypeNodeId = NodeId.Parse(EventTemplate.EventFilter.EventTypeDefinitionId);
+                var fields = ClientUtils.CollectInstanceDeclarationsForType(session, simpleEventTypeNodeId);
+
+                var eventFilter = new EventFilter();
+                foreach (var field in fields) {
+                    eventFilter.SelectClauses.Add(new SimpleAttributeOperand() {
+                        TypeDefinitionId = ObjectTypeIds.BaseEventType,
+                        BrowsePath = field.BrowsePath,
+                        AttributeId = (field.NodeClass == Opc.Ua.NodeClass.Variable) ? Attributes.Value : Attributes.NodeId
+                    });
+                }
+                eventFilter.WhereClause = new ContentFilter();
+                eventFilter.WhereClause.Push(FilterOperator.OfType, simpleEventTypeNodeId);
+
+                return eventFilter;
+            }
+
             private HashSet<uint> _newTriggers = new HashSet<uint>();
             private HashSet<uint> _triggers = new HashSet<uint>();
             private Publisher.Models.MonitoringMode? _modeChange;
             private readonly ILogger _logger;
         }
 
+        internal static class ClientUtils {
```

Comment:
Can we put this in a separate file?

Reply:
We can, I just followed the pattern with many classes inside SubscriptionServices. But I can move it out.

Reply:
Yeah, let's move it out. The file is already large.

Reply:
Fixed.